src/main/java/org/truffleruby/core/regexp/MatchDataNodes.java

```diff
@@ -203,16 +222,47 @@ public Object getIndex(DynamicObject matchData, int index, int length) {
             return createArray(store, length);
         }
 
+        @Specialization(guards = { "isRubySymbol(cachedIndex)", "name != null", "backRefs == 1",
+                "getRegexp(matchData) == regexp", "cachedIndex == index" })
+        public Object getIndexSymbolSingleMatch(DynamicObject matchData, DynamicObject index, NotProvided length,
+                @Cached("index") DynamicObject cachedIndex,
+                @Cached("getRegexp(matchData)") DynamicObject regexp,
+                @Cached("findNameEntry(regexp, index)") NameEntry name,
+                @Cached("numBackRefs(name)") int backRefs,
+                @Cached("backRefIndex(name)") int backRefIndex,
+                @Cached("createBinaryProfile()") ConditionProfile normalizedIndexProfile,
+                @Cached("createBinaryProfile()") ConditionProfile indexOutOfBoundsProfile,
+                @Cached("createBinaryProfile()") ConditionProfile hasValueProfile) {
+            return getIndex(matchData, backRefIndex, NotProvided.INSTANCE, normalizedIndexProfile, indexOutOfBoundsProfile, hasValueProfile);
```

Comment:
I recommend against calling directly other specializations methods, especially if they have profiles.
Instead, please create a abstract `executeGetIndex()` method and call it, and then these profiles are no longer needed.